src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java

```diff
@@ -254,7 +274,24 @@ private Result executeCmd(String cmd) throws IOException, InterruptedException {
                     .stderr(stderr)
                     .cmdAsSingleString(cmd)
                     .start().join() == 0;
-            String stdoutStr = stdout.toString();
+
+            // capture the stdout so it can be parsed later on
+            String stdoutStr = null;
+            try {
+                Computer computer = Computer.currentComputer();
+                if (computer != null) {
+                    Charset charset = computer.getDefaultCharset();
+                    if (charset != null) {
+                        stdoutStr = baos.toString(charset.name());
```

Comment:
IIUC, you may not have flushed the `stdout` stream. In fact the `join()` does not guarantee that `stdout` has finished pumping, you may only be reading the initial buffer's worth in `baos`